include/revng/ABI/Definition.h

```diff
@@ -154,6 +154,13 @@ type: struct
       this is apple flavor of the arm64 ABI.
     type: bool
 
+  - name: UseStrictAggregateAlignmentRules
+    doc: |
+      Treat all the structs passed using the stack as if they were aligned
+      unnaturally.
```

Comment:
1. Can you define "unnatural alignment"?
2. Can you make an example where this makes a difference?

Reply:
1. A struct that breaks natural alignment rules - as in contains at least one field for which the offset is not a multiple of its natural alignment.

2. Let me go into a bit of a tangent for context sake then. Technically both of these new options can be merged together since are only ever set to `true` for the apple arm64 abi. But I feel like they are distinct enough to be separate. Hopefully, for future reusal. the TL;DR is that `AllowPuttingSimilarlyAlignedStackArgumentsTogether` states that argument use relaxed alignment rules (as in they were normal fields of a struct). While `UseStrictAggregateAlignmentRules` states that "all the aggregates cannot be trusted, tip-toe around them and don't make any hard assumption", as in, let them behave the same way other unnatural struct behave, like, say
```cpp
struct __attribute__((packed)) {
  uint8_t field_at_offset_0;
  uint64_t unnaturally_aligned_field_at_offset_1;
};
```

Reply:
1. Can we put a micro glossary at the top of the file?
2. So AFAIU we're following rules that are more typical for laying out fields in a `struct` rather than laying out arguments? If so, maybe we should have a single option that reflects this?

Reply:
I think this too complex for me too understand without a specific example. Please make a specific example explaining the different behavior.
Each configuration option should come with an example, otherwise they are inaccessible.

Reply:
While at it I also moved the documentation out of that crowded `markdown` within `yaml` within `c++ header` structure into it's own file. Please give it a read when you have a moment. It's in `references/abi-definition.md`. Might also be a good time to elaborate on some of them that feel unclear to you.

Reply:
Oh God this is so good. Let me read it and leave some comment there. Thanks thanks thanks.